<!DOCTYPE html>
<html lang="en">

<head>
    <meta charset="UTF-8">
    <meta http-equiv="X-UA-Compatible" content="IE=edge">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Promise对象catch方法  </title>
</head>

<body>
    <script>
         const p = new Promise((resolve,reject)=>{
            setTimeout(() => {
                //设置 p 对象的状态为失败，并设置失败的值
                reject('出错啦。。。。。。。')
            }, 1000);
         })
        //  p.then(value=>{

        //  },reason=>{
        //     console.log(reason);
        //  })

        //  catch(onrejected?: (reason: any) => PromiseLike<never>): Promise<any>
         p.catch(reason=>{
            console.error(reason);
         })
    </script>
</body>

</html>